tools/astarte_e2e/lib/astarte_e2e/service_notifier.ex

```diff
@@ -104,33 +147,87 @@ defmodule AstarteE2E.ServiceNotifier do
   def service_down({:call, from}, :notify_service_up, data) do
     actions = [{:reply, from, :ok}]
 
-    Email.service_up_email()
+    failure_id =
+      data
+      |> get_failure_id()
+
+    updated_data =
+      data
+      |> reset_failures_before_alert()
+      |> reset_failure_id()
+
+    Email.service_up_email(failure_id)
     |> deliver()
 
-    Logger.info("Service up. The user has been notified.", tag: "service_up_notified")
-    {:next_state, :service_up, data, actions}
-  end
+    Logger.info("Service up. The user has been notified. Last FailureID: #{failure_id}.",
```

Comment:
put `failure_id` as metadata in the logger instead of printing it (remember to whitelist the metadata key in the logger configuration)